changelogs/fragments/79945-host_group_vars-improvements.yml

```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - Cache host_group_vars after loading it once since it is stateless and limit the amount of work it needs to do every time it runs.
```

Comment:
```suggestion
  - Cache host_group_vars after instantiating it once since it is stateless. Also limit the amount of repetitive work it needs to do every time it runs.
```